firefox-ios/Client/Frontend/Home/JumpBackIn/JumpBackInViewModel.swift

```diff
@@ -349,10 +350,14 @@ extension JumpBackInViewModel: HomepageViewModelProtocol {
     func refreshData(for traitCollection: UITraitCollection,
                      size: CGSize,
                      isPortrait: Bool = UIWindow.isPortrait,
-                     device: UIUserInterfaceIdiom = UIDevice.current.userInterfaceIdiom) {
+                     device: UIUserInterfaceIdiom = UIDevice.current.userInterfaceIdiom,
+                     orientation: UIDeviceOrientation = UIDevice.current.orientation) {
+        // UIDevice is not always returning the correct orientation so we check against the window orientation as well
```

Comment:
Ugh wacky! Why is this?

Reply:
`UIWindow.isPortrait` depends on `keyWindow` property where we access `UIApplication.shared.connectedScenes` and looks for a scene in the foreground and active state (.foregroundActive). On application startup if the window hierarchy is still being initialized `keyWindow` will return `nil`, causing `isPortrait` to return `false`.

I think `UIWindow.isPortrait` should be an optional instead of returning `false` when this use case happens. This would make it more implicit of the issue of using those properties. We use `UIWindow.isPortrait` in a bunch of places as default value so this would mean a bit of a refactor if we go that route.